Packages/com.unity.inputsystem/Tests/InputSystem/CoreTests_Actions.cs

```diff
@@ -1108,13 +1108,74 @@ public void Actions_WithMultipleBoundControls_DriveInteractionsFromControlWithGr
         }
     }
 
-    // There can be situations where two different controls are driven from the state. Most prominently, this is
+    [Test]
+    [Category("Actions")]
+    public void Actions_WithMultipleBoundControls_CanHandleInteractionsThatTriggerOnlyOnButtonRelease()
+    {
+        var keyboard = InputSystem.AddDevice<Keyboard>();
+        var gamepad = InputSystem.AddDevice<Gamepad>();
+
+        // Action with PressInteraction set to ReleaseOnly.
+        // In this configuration, PressInteraction will not do anything on button down. Make sure that
+        // the disambiguation code does not get confused by this but still correctly processes the button
+        // up state change (which is a state change *back* to default state, i.e. has zero magnitude).
+        var action = new InputAction(interactions: "Press(behavior=1)");
+
+        action.AddBinding("<Keyboard>/a");
+        action.AddBinding("<Gamepad>/buttonNorth");
+
+        action.Enable();
+
+        using (var trace = new InputActionTrace())
+        {
+            trace.SubscribeToAll();
+
+            Press(gamepad.buttonNorth);
+
+            Assert.That(trace, Is.Empty);
+
+            Release(gamepad.buttonNorth);
+
+            var actions = trace.ToArray();
+            Assert.That(actions, Has.Length.EqualTo(1));
+            Assert.That(actions[0].phase, Is.EqualTo(InputActionPhase.Performed));
+            Assert.That(actions[0].control, Is.SameAs(gamepad.buttonNorth));
+
+            trace.Clear();
+
+            Press(gamepad.buttonNorth);
+
+            Assert.That(trace, Is.Empty);
+
+            Release(gamepad.buttonNorth);
+
+            actions = trace.ToArray();
+            Assert.That(actions, Has.Length.EqualTo(1));
+            Assert.That(actions[0].phase, Is.EqualTo(InputActionPhase.Performed));
+            Assert.That(actions[0].control, Is.SameAs(gamepad.buttonNorth));
```

Comment:
Why do we test this case twice?

Reply:
Probably warrants a comment. I had the case where the code wouldn't trigger correctly again for the same button so wanted to make sure that the second button press of the same button comes through correctly.